### PR TITLE
Ab login user token no

### PR DESCRIPTION
**Pull request by Abiidahiya**

### CODE REVIEW

File: lib/utils/app_string_res.dart [hunk 1]
```diff
@@ -27,6 +27,7 @@ const lastnameerror='Please enter a valid last name';
 
 const emailerror='Please enter a valid email';
 const address='Enter address manually';
+const addresserror='Please enter a valid address';
```
Comment (deepakkanyan) [inline lib/utils/app_string_res.dart]: make all string constant readable : 
like 
`const pls_enter_valid_address ='Please enter a valid address';`

Comment (deepakkanyan) [reply lib/utils/app_string_res.dart]: try to make variable value more readable

Comment (Abiidahiya) [reply lib/utils/app_string_res.dart]: ok deepak ji

File: lib/utils/firebase_services.dart [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      return false; // No user is signed in.
+    } catch (e) {
+      print('Error uploading data to Firestore: $e');
+      return false;
+    }
+
+  }
+  Future<bool> updateUserBio(String userFirebaseUid, String userBio) async {
+    try {
+      final userRef = _firestore.collection('users').doc(userFirebaseUid);
```
Comment (deepakkanyan) [inline lib/utils/firebase_services.dart]: Keep all table and database in new Constant Files

appdatabase/model/DbConstants

File: lib/utils/firebase_services.dart [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    }
+  }
+  Future<String?> updateUserPhotoURL(File imageFile, String userFirebaseUid) async {
+    try {
+      // Upload the image to Firebase Storage and get the download URL
+      final photoURL = await _storageService.uploadImage(imageFile, userFirebaseUid);
+
+      if (photoURL != null) {
+        // Update the user's document in Firestore with the new photoURL
+        final userRef = _firestore.collection('users').doc(userFirebaseUid);
```
Comment (deepakkanyan) [inline lib/utils/firebase_services.dart]: same here.

File: lib/utils/firebase_storage_services.dart [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import 'dart:io';
+import 'package:firebase_storage/firebase_storage.dart';
+
+class FirebaseStorageService {
+  final FirebaseStorage _storage = FirebaseStorage.instance;
+
+  Future<String?> uploadImage(File imageFile, String userFirebaseUid) async {
+    try {
+      final fileName = '$userFirebaseUid-${DateTime.now().millisecondsSinceEpoch}.jpg';
+      final storageRef = _storage.ref().child('user_photos').child(fileName);
```
Comment (deepakkanyan) [inline lib/utils/firebase_storage_services.dart]: same here

File: lib/utils/validators.dart [hunk 5]
```diff
@@ -48,4 +48,17 @@ class Validators {
 
     return null;
   }
-}
+  static String? addressValidator(String? value) {
```
Comment (deepakkanyan) [inline lib/utils/validators.dart:51]: test Case for this function?

File: lib/view/ui/userdatawizard/personalinfo_page.dart [hunk 6]
```diff
@@ -16,14 +20,26 @@ class PersonalInfoPage extends StatelessWidget {
...
+  final FirebaseService _firebaseService = FirebaseService();
+  final FirebaseStorageService storageService = FirebaseStorageService();
+
+  Map<String, dynamic> collectUserData() {
+    final firstName = controller.firstNameController.text;
+    final lastName = controller.lastNameController.text;
+    final email = controller.emailController.text;
+    final displayName = '$firstName $lastName'; // Combine first and last name
+
+
```
Comment (deepakkanyan) [inline lib/view/ui/userdatawizard/personalinfo_page.dart:33]: it should retrun data Model class like 

class USerData{
String  fname, String lastname
}

File: lib/view/ui/userdatawizard/personalinfo_page.dart [hunk 7]
```diff
@@ -16,14 +20,26 @@ class PersonalInfoPage extends StatelessWidget {
...
+  }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(
-        title: const Text(personalinfomessage),
-        backgroundColor: Colors.blueAccent,
-      ),
+      appBar: null,
```
Comment (deepakkanyan) [inline lib/view/ui/userdatawizard/personalinfo_page.dart:43]: can we remove this null or remove appBar.

File: lib/view/ui/userdatawizard/personalinfo_page.dart [hunk 8]
```diff
@@ -65,10 +81,65 @@ class PersonalInfoPage extends StatelessWidget {
...
+                    if (user != null) {
+                      userFirebaseUid = user.uid;
+
+                      // Check if an image has been selected
+                      if (controller.selectedImage.value != null) {
+                        final photoURL = await _firebaseService.updateUserPhotoURL(controller.selectedImage.value!, userFirebaseUid);
+                        if (photoURL != null) {
+                          userData['photoURL'] = photoURL; // Update userData with photoURL
+                        } else {
+                          // Handle the case where image upload fails
```
Comment (deepakkanyan) [inline lib/view/ui/userdatawizard/personalinfo_page.dart]: make Toast Message gobal

File: lib/view/ui/userdatawizard/personalinfo_page.dart [hunk 9]
```diff
@@ -65,10 +81,65 @@ class PersonalInfoPage extends StatelessWidget {
               CustomSizedBox(),
               ElevatedButton(
                 key: Key('nextButton'),
-                onPressed: () {
+                style: ElevatedButton.styleFrom(
+                  backgroundColor: Theme.of(context).colorScheme.secondary,
+                ),
+                onPressed: () async {
                   if (controller.formKey.currentState!.validate()) {
```
Comment (deepakkanyan) [inline lib/view/ui/userdatawizard/personalinfo_page.dart]: break this on Press code in  funtions.

Alway break code to multple funtion, if multple lines are in same funtions

File: lib/view/ui/userdatawizard/personalinfo_page.dart [hunk 10]
```diff
@@ -65,10 +81,65 @@ class PersonalInfoPage extends StatelessWidget {
...
+                          return; // Don't proceed if image upload fails
+                        }
+                      }
+
+                      // Pass the userFirebaseUid and updated userData to uploadUserData
+                      final success = await _firebaseService.uploadUserData(userData);
+
+                      if (success) {
+                        // Data uploaded successfully
+                        ScaffoldMessenger.of(context).showSnackBar(
```
Comment (deepakkanyan) [inline lib/view/ui/userdatawizard/personalinfo_page.dart]: same here, Always check for dublicate code.,DO not repeat code.

File: lib/view/ui/userdatawizard/personalinfo_page.dart [hunk 11]
```diff
@@ -65,10 +81,65 @@ class PersonalInfoPage extends StatelessWidget {
...
+                          SnackBar(
+                            content: Text('Data uploaded successfully'),
+                            duration: Duration(seconds: 2),
+                          ),
+                        );
+                        wizardController.goToNextStep();
+                      } else {
+                        // Handle the case where data upload fails
+                        ScaffoldMessenger.of(context).showSnackBar(
```
Comment (deepakkanyan) [inline lib/view/ui/userdatawizard/personalinfo_page.dart]: Same here

File: lib/view/ui/userdatawizard/personalinfo_page.dart [hunk 12]
```diff
@@ -65,10 +81,65 @@ class PersonalInfoPage extends StatelessWidget {
...
+                          SnackBar(
+                            content: Text('Failed to upload data'),
+                            duration: Duration(seconds: 2),
+                          ),
+                        );
+                      }
+                    } else {
+                      // Handle the case where no user is signed in
+                      ScaffoldMessenger.of(context).showSnackBar(
```
Comment (deepakkanyan) [inline lib/view/ui/userdatawizard/personalinfo_page.dart]: Same here

File: lib/view/ui/userdatawizard/userbio_page.dart [hunk 13]
```diff
@@ -30,22 +32,48 @@ class UserBioPage extends StatelessWidget {
...
+                    } else {
+                      // Handle the case where data upload fails
+                    }
+                  } else {
+                    // Handle the case where no user is signed in
+                  }
                 }
               },
-              child: Text(submit),
+              child: Text('Submit'),
```
Comment (deepakkanyan) [inline lib/view/ui/userdatawizard/userbio_page.dart]: constant?

File: lib/view/ui/userprofile/profile_page.dart [hunk 14]
```diff
@@ -16,31 +19,109 @@ class ProfilePage extends StatefulWidget {
...
+  void initState() {
+    super.initState();
+    _getUserData();
+  }
+
+  Future<void> _getUserData() async {
+    _user = _auth.currentUser;
+    if (_user != null) {
+      final uid = _user!.uid;
+      final userDoc = await _firestore.collection('users').doc(uid).get();
```
Comment (deepakkanyan) [inline lib/view/ui/userprofile/profile_page.dart:40]: Table constant?

File: lib/view/ui/userprofile/profile_page.dart [hunk 15]
```diff
@@ -16,31 +19,109 @@ class ProfilePage extends StatefulWidget {
...
+                decoration: BoxDecoration(
+                  color: Theme.of(context).colorScheme.primary, // Change the color as needed
+                  borderRadius: BorderRadius.circular(10),
+                ),
+                child: Center(
+                  child: _userData != null
+                      ? Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      CircleAvatar(
```
Comment (deepakkanyan) [inline lib/view/ui/userprofile/profile_page.dart:82]: Need to change good UI. [will do it later]

File: lib/view_model/userlogin/sign_google.dart [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+      print('photoURL: $photoURL');
+
+      final GoogleSignInAuthentication googleAuth =
+      await googleUser.authentication;
+      final AuthCredential credential = GoogleAuthProvider.credential(
+        accessToken: googleAuth.accessToken,
+        idToken: googleAuth.idToken,
+      );
+
+      UserCredential userCredential =
```
Comment (deepakkanyan) [inline lib/view_model/userlogin/sign_google.dart:31]: usse Table constant.

File: pubspec.yaml [hunk 17]
```diff
@@ -38,6 +38,7 @@ dependencies:
   location: ^5.0.0
   image_picker: ^1.0.2
   flutter_wizard: ^2.0.5
+  uuid: ^4.0.0
```
Comment (deepakkanyan) [inline pubspec.yaml]: remove this 'uuid' lib,

File: assets/images/ic_blank_profilepic.jpg
Comment (deepakkanyan) [inline assets/images/ic_blank_profilepic.jpg:1]: we need 2 users place holder image ?

File: lib/utils/firebase_services.dart (hunk 2 above)
Comment (Abiidahiya) [reply lib/utils/firebase_services.dart]: Should I keep them in one file ie firebase_services.dart or creat a seperate file for each required method?

Comment (deepakkanyan) [reply lib/utils/firebase_services.dart]: it should ne new folder,
  appdatabase/tables/UserTableConstant

File: assets/images/ic_blank_profilepic.jpg
Comment (Abiidahiya) [reply assets/images/ic_blank_profilepic.jpg:1]: No, we need just one.. I will delete the other one. It was just for comparison which one looks better in the app

File: lib/utils/validators.dart (hunk 5 above)
Comment (Abiidahiya) [reply lib/utils/validators.dart:51]: Will update

File: lib/view/ui/userdatawizard/personalinfo_page.dart (hunk 6 above)
Comment (Abiidahiya) [reply lib/view/ui/userdatawizard/personalinfo_page.dart:33]: yes the model class final userData = collectUserData(); is there in the onpressed callback function and the function collectUserData returns as 
 return {
      'displayName': displayName,
      'email': email,
    };

File: pubspec.yaml [hunk 18]
```diff
@@ -30,15 +30,19 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  cloud_firestore: ^4.8.2
-  firebase_storage: ^11.1.2
+  cloud_firestore: ^4.9.1
+  firebase_storage: ^11.2.6
   flutter_facebook_auth: ^6.0.0
```
Comment (deepakkanyan) [inline pubspec.yaml:35]: do we need it?

Comment (Abiidahiya) [reply pubspec.yaml:35]: yes, we need the firestore cloud for the database, and firebase storage to upload and store the image file to get a URL and that URL will be saved in the database.

File: lib/view/ui/userdatawizard/personalinfo_page.dart (hunk 8 above)
Comment (Abiidahiya) [reply lib/view/ui/userdatawizard/personalinfo_page.dart]: should i make string constants to be used in the toast or i should let it remain like this and enter data in string??

**Pull request merged**